pydoctor/napoleon/docstring.py

```diff
@@ -0,0 +1,1617 @@
+"""
```

Comment:
I suggest we don't take too much time to review this file as it's pretty huge and most of it's code is coming straight from sphinx's extension "napoleon".

Reply:
Nevertheless, I’ve donne some major improvements to the original version of napoleon Sphinx extension, keeping the same behavior (This is tested). I’m pretty familiar with it now. I can ensure you that, if there is a bug/improvement to apply, I’ll look at it pretty quickly.

What I’m saying is that,  it’s a big file and to do an exhaustive review of it would probably take too much time any of the current reviewers can spend. Then, this PR will stay unmerged for ever. This would be a shame, so I’de suggest we skip exhaustive review for this particular case (I did not went yolo while coding this). Unless, we risk to never offer this feature that is google and numpy docstring parsing.

Let’s note that at least one project is using my development branch with numpy parsing to have this feature (

 Tell me what you think @adiroiban @mthuurne